### PR TITLE
Respect `order`

### PR DESCRIPTION
**Pull request by trzysiek**
Checked the dashboards at the end, and they seem still fine.

### CODE REVIEW

File: quesma/model/select.go [hunk 1]
```diff
@@ -11,11 +11,14 @@ type SelectCommand struct {
 	GroupBy     []Expr        // if not empty, we do GROUP BY GroupBy...
 	OrderBy     []OrderByExpr // if not empty, we do ORDER BY OrderBy...
 
-	Limit       int // LIMIT clause, noLimit (0) means no limit
-	SampleLimit int // LIMIT, but before grouping, 0 means no limit
+	LimitBy     []Expr // LIMIT BY clause (empty => maybe LIMIT, but no LIMIT BY)
+	Limit       int    // LIMIT clause, noLimit (0) means no limit
+	SampleLimit int    // LIMIT, but before grouping, 0 means no limit
+
+	Subqueries []SelectCommand
```
Comment (pdelewski) [inline quesma/model/select.go:18]: What kind of query do you want to express with `Subqueries`? Currently, we have `Columns`, `FromClause` and `WhereClause`, all of them are expressions and can be subqueries

Comment (trzysiek) [reply quesma/model/select.go:18]: I do a lot of
```
WITH subquery_1 AS (
  SELECT ...
), subquery_2 AS (
  SELECT ...
)
SELECT ...
FROM table INNER JOIN subquery_1 INNER JOIN subquery_2
```

Comment (pdelewski) [reply quesma/model/select.go:18]: ok, then maybe better name will be `CTE` - `Common Table Expressions` or `WithClause`

Comment (trzysiek) [reply quesma/model/select.go:18]: great, thanks! I'll change to CTE

File: quesma/queryparser/aggregation_parser.go [hunk 2]
```diff
@@ -766,64 +754,111 @@ func (cw *ClickhouseQueryTranslator) tryBucketAggregation(currentAggr *aggrQuery
...
+							mainOrderBy = cw.findMetricAggregation(queryMap, key, currentAggr)
+						}
+
+						fullOrderBy = []model.OrderByExpr{
+							{Exprs: []model.Expr{mainOrderBy}, Direction: direction, ExchangeToAliasInCTE: true},
+							{Exprs: []model.Expr{fieldExpression}},
+						}
+					}
+				} else {
+					logger.WarnWithCtx(cw.Ctx).Msgf("order has more than 1 key, but %d. Order: %+v. Using default", len(order), order)
```
Comment (jakozaur) [inline quesma/queryparser/aggregation_parser.go:842]: Not a small issue, I would use `ErrorWithCtxAndReason`.

File: quesma/queryparser/aggregation_parser.go [hunk 3]
```diff
@@ -766,64 +754,111 @@ func (cw *ClickhouseQueryTranslator) tryBucketAggregation(currentAggr *aggrQuery
...
+				}
+			} else {
+				logger.WarnWithCtx(cw.Ctx).Msgf("order is not a map, but %T, value: %v. Using default order", orderRaw, orderRaw)
```
Comment (jakozaur) [inline quesma/queryparser/aggregation_parser.go:845]: Feels like an error too.

File: quesma/queryparser/aggregation_parser.go [hunk 4]
```diff
@@ -766,64 +754,111 @@ func (cw *ClickhouseQueryTranslator) tryBucketAggregation(currentAggr *aggrQuery
...
+		}
+
+		currentAggr.Type = bucket_aggregations.NewTerms(cw.Ctx, termsType == "significant_terms", mainOrderBy)
+		currentAggr.SelectCommand.Limit = size
+		currentAggr.SelectCommand.Columns = append(currentAggr.SelectCommand.Columns, fieldExpression)
+		currentAggr.SelectCommand.GroupBy = append(currentAggr.SelectCommand.GroupBy, fieldExpression)
+		currentAggr.SelectCommand.LimitBy = append(currentAggr.SelectCommand.LimitBy, fieldExpression)
+		currentAggr.SelectCommand.OrderBy = append(currentAggr.SelectCommand.OrderBy, fullOrderBy...)
+		if missingPlaceholder == nil {
+			currentAggr.whereBuilder = model.CombineWheres(cw.Ctx, currentAggr.whereBuilder,
```
Comment (jakozaur) [inline quesma/queryparser/aggregation_parser.go:856]: Ok, would add comment to replace it with schema. In our semantic it is either NULL or default value (if not nullable type).

**Review by jakozaur — APPROVED** [review]
Minor comments, feel free to merge once you fix them.

**Pull request merged**